Services/Administration/OJS.Services.Administration.Data/Implementations/ProblemGroupsDataService.cs

```diff
@@ -24,6 +24,10 @@ public ProblemGroupsDataService(DbContext problemGroups)
             this.DbSet
                 .Where(pg => pg.ContestId == contestId);
 
+        public IQueryable<ProblemGroup> GetAllNonDeletedByContest(int contestId)
+            => this.GetAllByContest(contestId)
+                .Where(pg => !pg.IsDeleted);
+
```

Comment:
`NonDeleted` is not necessary, because we should get these by default. In case we get `WithDeleted`, then we should specify this into the name.